File: friendly_cars/src/main/java/com/friendlyCarsSystem/friendly_cars/service/ClientService.java
package com.friendlyCarsSystem.friendly_cars.service;

import java.util.List;
import java.util.Map;

import com.friendlyCarsSystem.friendly_cars.entity.Client;
import com.friendlyCarsSystem.friendly_cars.exception.ClientNotFoundException;

import org.springframework.http.ResponseEntity;

/**
 * ClientService
 */
public interface ClientService {
    List<Client> getAllClients();
    Client getClientById(String clientId) throws ClientNotFoundException;
    Client createClient(Client client);
    Client updateClient(String clientId, Client updatedClient) throws ClientNotFoundException;
    Client partialUpdateClient(String clientId, Map<Object, Object> fields) throws ClientNotFoundException;
    ResponseEntity<String> deletClient(String clientId) throws ClientNotFoundException;
}

File: friendly_cars_frontend/src/service/ClientService.js
import axios from "axios";

const CLIENT_BASE_URL = "http://localhost:8080/api/v1/clients";

class ClientService {
    createClient(client) {
        return axios.post(CLIENT_BASE_URL, client)
            .then(response => response.data)
    }

    getAllClients() {
        return axios.get(CLIENT_BASE_URL)
            .then(response => response.data)
    }

    getClientById(clientId) {
        return axios.get(CLIENT_BASE_URL + `/${clientId}`)
            .then(response => response.data)
    }

    updateClient(clientId, newClient) {
        return axios.put(CLIENT_BASE_URL + `/${clientId}`, newClient)
            .then(response => response.data)
    }

    partialUpdateClient(clientId, newClient) {
        return axios.patch(CLIENT_BASE_URL + `/${clientId}`, newClient)
            .then(response => response.data)
    }

    deleteClient(clientId) {
        return axios.delete(CLIENT_BASE_URL + `/${clientId}`)
            .then(response => response.data)
    }
}

export default new ClientService()
